interfaces/devdoc/srw_lock_ll_requirements.md

```diff
@@ -0,0 +1,88 @@
+# `srw_lock_ll` requirements
+
+## Overview
+
+`srw_lock_ll` is the implementation of a slim reader writer lock.
+
+## Exposed API
+
+```c
+#define SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES \
+    SRW_LOCK_LL_TRY_ACQUIRE_OK, \
+    SRW_LOCK_LL_TRY_ACQUIRE_COULD_NOT_ACQUIRE, \
+    SRW_LOCK_LL_TRY_ACQUIRE_INVALID_ARGS
+
+typedef void* SRW_LOCK_LL;
+
+MU_DEFINE_ENUM(SRW_LOCK_LL_TRY_ACQUIRE_RESULT, SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES)
+
+MOCKABLE_FUNCTION(, void, srw_lock_ll_init);
```

Comment:
>init [](http://example.com/codeflow?start=38&length=4)

poetry usually is with "create/destroy"unless it initializes global objects.

I don't see a problem having this as "_create" and return `void`, just saying.

Reply:
I think init/deinit fits better with the current terminology we're using. When we would like to initialize the memory provided by the caller we usually use init/deinit. DList as example does the same.

If you feel very strongly I can rename to create/destroy but I think we never had issues with poetry ...